Documentation/guides/MSBuildBestPractices.md

```diff
@@ -32,6 +32,77 @@ properties, targets, etc., as we would likely need to support them
 into oblivion. However, we might choose to "safely" deprecate them
 in a way that makes sense.
 
+## Item Group Transforms
+
+MSBuild has a widely used feature where you can create a one-to-one
+mapping from an `<ItemGroup/>` to a new `<ItemGroup/>`. The syntax for
+this looks like:
+
+```xml
+<ItemGroup>
+  <_DestinationFiles Include="@(_SourceFiles->'$(SomeDirectory)%(Filename)%(Extension)')" />
+</ItemGroup>
+```
+
+This takes a list of files, and creates a desired destination path for
+each file in `$(SomeDirectory)`. The `%(Filename)` and `%(Extension)`
+item metadata is used to get the filename of the source file. See [the
+MSBuild documentation on transforms][msbuild-transforms] for more
+info.
+
+One thing to note here, is we shouldn't have multiple transforms
+within the same target:
+
+```xml
+<Target Name="_CopyPdbFiles"
+    Inputs="@(_ResolvedPortablePdbFiles)"
+    Outputs="$(_AndroidStampDirectory)_CopyPdbFiles.stamp"
+    DependsOnTargets="_ConvertPdbFiles">
+  <CopyIfChanged
+      SourceFiles="@(_ResolvedPortablePdbFiles)"
+      DestinationFiles="@(_ResolvedPortablePdbFiles->'$(MonoAndroidLinkerInputDir)%(Filename)%(Extension)')"
+  />
+  <Touch Files="$(_AndroidStampDirectory)_CopyPdbFiles.stamp" AlwaysCreate="True" />
+  <ItemGroup>
+    <FileWrites Include="@(_ResolvedPortablePdbFiles->'$(MonoAndroidLinkerInputDir)%(Filename)%(Extension)')" />
+  </ItemGroup>
+</Target>
+```
+
+Running this transformation twice:
+
+    @(_ResolvedPortablePdbFiles->'$(MonoAndroidLinkerInputDir)%(Filename)%(Extension)')
```

Comment:
Unrelated: it is *amusing* that GitHub's `diff` viewer for Markdown gets confused here, and interprets the `_` in the above line as the start of *italics*, so *everything afterward* is italic...in the *diff* view.  The subsequent `xml` block also loses syntax coloring, unlike the preceding `xml` block.

The actual page rendering is fine.

I wonder if GitHub would show this diff better if a TAB was used instead of 4 spaces?